test/libyul/StackShufflingTest.cpp

```diff
@@ -14,41 +14,172 @@
     You should have received a copy of the GNU General Public License
     along with solidity.  If not, see <http://www.gnu.org/licenses/>.
 */
-/**
- * Unit tests for stack shuffling.
- */
+
+#include <test/libyul/StackShufflingTest.h>
+
+#include <liblangutil/Scanner.h>
+#include <libsolutil/AnsiColorized.h>
 #include <libyul/backends/evm/StackHelpers.h>
-#include <boost/test/unit_test.hpp>
 
-using namespace std;
+using namespace solidity::util;
 using namespace solidity::langutil;
+using namespace solidity::yul;
+using namespace solidity::yul::test;
+using namespace std;
 
-namespace solidity::yul::test
+bool StackShufflingTest::parse(string const& _source)
 {
+	CharStream stream(_source, "");
+	Scanner scanner(stream);
 
-BOOST_AUTO_TEST_SUITE(YulStackShuffling)
+	auto readFunctionCall = [](Scanner* scanner) -> yul::FunctionCall
```

Comment:
```suggestion
	auto readFunctionCall = [](Scanner& scanner) -> yul::FunctionCall
```

The general C++ rule-of-thumb on this is: always use references (i.e. ``Scanner&``), if this will *always* have a valid value (which is the case here) - and only use pointers (i.e. ``Scanner*``), if there are situations in which you want to pass ``nullptr`` (which you then check against explicitly).

There's some exceptions to that rule (for example you can't store references in all kinds of containers, since that sometimes involves having a default/uninitialized value and such), but here it should be just fine to use a reference.